### DISCUSSION TITLE
Does this library support the idea of sending workouts to the PM5?

### DISCUSSION THREAD

**Discussion by avanlaningham** [tijmenvangulik/ErgometerJS#26] [Q&A]
So it would create all the intervals on the PM5 itself?  I tried to look through code and documentation but this was not clear to me.  We would love to use this for our app that now supports PM5  https://trainerday.com  We are currently primarily cycling focused and the largest cycling structured workout and plan sharing platform but moving to SwimERG  (By Vasa) and RowERG (C2).  We are happy to contribute to this library if needed.  It seems like development has paused.

**Comment by tijmenvangulik** [comment]
Some commands can be called by every one and some are proprietary.  You have to check the (latest) docs on which commands are made publicly available. For example I could call the setProgram without any issues. To be able to call the proprietary commands you will have tho contact concept 2 to learn the secret hand shake. (I know it is possible because time team uses my library and uses functionality of the PM which can only be accused if they use proprietary commands )

More on info is in this discussion (also an attachment with the concept 2 docs )

https://github.com/tijmenvangulik/ErgometerJS/discussions/21

Ergometer-js has simplified ways of sending commands but you can also send raw commands which you find in the concept2 documentation.

**Comment by tijmenvangulik** [comment]
Adding workouts is now added. It only works for blue tooth because for USB you need an authorization from concept2
